### ISSUE TITLE
macos_application not honoring minimum_os_version

### ISSUE THREAD

**Issue by mattryavec** [bazelbuild/rules_apple#126]
MacOS apps built via the `macos_application` rule appear to be using the SDK version as the target deployment version, rather than the specified `minimum_os_version` in the build rule. This can be reproduced with the example MacOS HelloWorld app in this repo.

Replication steps:
1) Build `//examples/macos/HelloWorld`, which has the minimum_os_version set to 10.11, using a later SDK version (e.g, 10.13)
2) Attempt to run the app on a machine running a MacOS lower than the SDK version (10.11 or 10.12 in this case)

Actual result: The app does not open, and a dialog message is given noting that the app must be run on 10.13 or later.

Expected: The app opens

**Comment by joprice** [comment]
I have `minimum_os_version` set to `10.10`. Running `otool -l` and looking for `LC_VERSION_MIN_MACOSX` shows:

```
cmd LC_VERSION_MIN_MACOSX
  cmdsize 16
  version 10.12
      sdk 10.13
```

**Comment by joprice** [comment]
Setting MACOSX_DEPLOYMENT_TARGET fixed this for me:

`build --action_env MACOSX_DEPLOYMENT_TARGET=10.10`

**Comment by keith** [comment]
This is fixed in bazel and will be released as part of 0.23.0 https://github.com/bazelbuild/bazel/pull/6891

**Comment by sergiocampama** [comment]
Closing since bazel 0.23 has been released
